tests/test_google_ads_field_exclusion.py

```diff
@@ -0,0 +1,204 @@
+                for field, values in field_exclusions.items():
+                    if values == []:
+                        fields_without_exclusions.append(field)
+
+                # Gather fields with exclusions as input to randomly build maximum length selection set
+                fields_with_exclusions = []
+                for field, values in field_exclusions.items():
+                    if values != []:
+                       fields_with_exclusions.append(field)
+
+                if len(fields_with_exclusions) == 0:
+                    raise AssertionError(f"Skipping assertions. No field exclusions for stream: {stream}")
+
+                self.stream = stream
+                random_order_of_exclusion_fields[stream] = []
+                self.random_order_of_exclusion_fields = random_order_of_exclusion_fields
+
+                random_exclusion_field_selection_list = self.random_field_gather(fields_with_exclusions)
+                field_selection_set = set(random_exclusion_field_selection_list + fields_without_exclusions)
+
+                with self.subTest(order_of_fields_selected=self.random_order_of_exclusion_fields[stream]):
+
+                    # Select fields and re-pull annotated_schema.
+                    self.select_stream_and_specified_fields(conn_id, catalogs_to_test[0], field_selection_set)
+
+                    try:
+                        # Collect updated metadata
+                        schema_2 = menagerie.get_annotated_schema(conn_id, catalogs_to_test[0]['stream_id'])
+
+                        # Verify metadata for all fields
+                        for rec in schema_2['metadata']:
+                            if rec['breadcrumb'] != [] and rec['breadcrumb'][1] != "_sdc_record_hash":
+                                # Verify metadata for selected fields
+                                if rec['breadcrumb'][1] in field_selection_set:
+                                    self.assertEqual(rec['metadata']['selected'], True,
+                                                msg="Expected selection for field {} = 'True'".format(rec['breadcrumb'][1]))
+
+                                else:  # Verify metadata for non selected fields
+                                    self.assertEqual(rec['metadata']['selected'], False,
+                                                     msg="Expected selection for field {} = 'False'".format(rec['breadcrumb'][1]))
+
+                        # Run a sync
+                        sync_job_name = runner.run_sync_mode(self, conn_id)
+                        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+
+                        self.assertEqual(0, exit_status.get('tap_exit_status'))
+                        self.assertEqual(0, exit_status.get('target_exit_status'))
+                        self.assertEqual(0, exit_status.get('discovery_exit_status'))
+                        self.assertIsNone(exit_status.get('check_exit_status'))
```

Comment:
I think these could be replaced with the `run_and_verify_sync` method in base.

Reply:
Used menagerie verify.